controller/engine_controller.go

```diff
@@ -580,6 +622,20 @@ func (ec *EngineController) DeleteInstance(obj interface{}) (err error) {
 		}
 	}
 
+	// If the node is unreachable, don't bother with the successive timeouts we would spend attempting to contact
+	// its client proxy to delete the engine.
+	if isRWXVolume {
+		isDelinquent, _ := ec.ds.IsNodeDelinquent(im.Spec.NodeID, e.Spec.VolumeName)
+		if isDelinquent {
+			log.Infof("Skipping deleting RWX engine %v since IM node %v is delinquent", e.Name, im.Spec.NodeID)
+			if e.Spec.NodeID != "" {
+				log.Infof("Clearing delinquent nodeID for RWX engine %v", e.Name)
+				e.Spec.NodeID = ""
+			}
+			return nil
+		}
+	}
```

Comment:
I introduced the logic so that we attempt to delete engine process on the delinquent node at least ONCE. But ignore the error if the deletion can't succeded (this is to unlock the volume so it is not stuck on detaching state forever on the down node and can move to the good node).

Is there a chance of orphan process if the node is not actually dead? Yes, there is a change. We will handle it similar to how we handle a regular RWX volume's engine deletetion here https://github.com/longhorn/longhorn-manager/blob/224fa04290ac9daede00b565b1598b7c198d5a90/controller/engine_controller.go#L589-L604

WDYT? 

cc @james-munson @ejweber

Reply:
Sounds good!